caravel/models.py

```diff
@@ -154,42 +154,37 @@ class Slice(Model, AuditMixinNullable):
     __tablename__ = 'slices'
     id = Column(Integer, primary_key=True)
     slice_name = Column(String(250))
-    druid_datasource_id = Column(Integer, ForeignKey('datasources.id'))
-    table_id = Column(Integer, ForeignKey('tables.id'))
+    datasource_id = Column(Integer)
     datasource_type = Column(String(200))
     datasource_name = Column(String(2000))
     viz_type = Column(String(250))
     params = Column(Text)
     description = Column(Text)
     cache_timeout = Column(Integer)
     perm = Column(String(2000))
-
-    table = relationship(
-        'SqlaTable', foreign_keys=[table_id], backref='slices')
-    druid_datasource = relationship(
-        'DruidDatasource', foreign_keys=[druid_datasource_id], backref='slices')
     owners = relationship("User", secondary=slice_user)
 
     def __repr__(self):
         return self.slice_name
 
     @property
+    def cls_model(self):
+        return src_registry.get_cls_model(self.datasource_type)
+
+    @property
```

Comment:
I kept the property and added a getter method for this property with memorized decorator. I'm not sure why, but the decorator makes the function to return functools.partial without a getter or property decorator. That's why I did it this way. I believe this will server the same purpose, which is reduce the number of times a request is being made.